### ISSUE TITLE
[C4GT Community]: Replace FromOriginMatchHeader secret to locally set environment variable

### ISSUE THREAD

**Issue by codecakes** [Xcov19/project-healthcare#57]
## Description
There is a [binder](https://www.neoteroi.dev/blacksheep/requests/#reading-parameters-from-the-request-object) `FromHeader` here:

https://github.com/Xcov19/project-healthcare/blob/e8fb0a6a8a6cbcb0f7fd1b8f7429790eadea2930/xcov19/app/settings.py#L48

and here:
https://github.com/Xcov19/project-healthcare/blob/e8fb0a6a8a6cbcb0f7fd1b8f7429790eadea2930/xcov19/app/middleware.py#L20

which are currently hardcoded. It needs to be replaced with a value sourced directly from environment variable. The value can be any sophisticated super secret string one can conjure in the localhost.

The code describes where and why this is used. 

Reference: You see this getting used during API calls here: https://project-healthcare.onrender.com/docs

## Goals
- [ ] Refactor to use locally generated from environment variable
- [ ] `make run` and smoke test to verify APIs don't break after refactor

### Expected Outcome

- [ ] No change in API behaviour
- [ ] Sending wrong secret in the API call which is different from locally generated secret fails

### Acceptance Criteria

There is a Pytest e2e Testsuite which tests for the expected outcomes in the following cases:

- [ ] Test asserts true when no change in API behaviour when correct secret is supplied
- [ ] Test asserts failure when sending wrong secret in the API call which is different from locally generated secret fails
- [ ] No mocks are used.

See https://tonybaloney.github.io/posts/async-test-patterns-for-pytest-and-unittest.html

### Implementation Details

1. Ensure locally sourced env var is set in `settings.py``
2. It is then set in https://github.com/Xcov19/project-healthcare/blob/e8fb0a6a8a6cbcb0f7fd1b8f7429790eadea2930/xcov19/app/settings.py#L48
3. The placeholder string is replaced in relevant places with locally set env variable

### Product Name

project-healthcare

### Organisation Name

XCoV19

### Domain

Healthcare

### Tech Skills Needed

CI/CD, Debugging, Python, RESTful APIs, Swagger, Test, Testing Library

### Mentor(s)

@codecakes 

### Complexity

Medium

### Category

API, Backend, Beginner Friendly, CI/CD, Testing

**Comment by TheSciPro** [comment]
Hey,
Is this issue still open?

**Comment by codecakes** [comment]
> Hey, Is this issue still open?

@TheSciPro 
Yes. you can check all issues by their [status](https://github.com/Xcov19/project-healthcare/issues?q=is%3Aopen+is%3Aissue)

**Comment by MustafaAkolawala** [comment]
Hello,
can I get this issue assigned to me

**Comment by prajwal2431** [comment]
assign this issue to me

**Comment by codecakes** [comment]
@prajwal2431 @TheSciPro @MustafaAkolawala if you follow the necessary actions asked in this issue, you are free to raise a PR. It is a on a first-come first-serve basis. Some general pointers to follow when implementing PR for this issue are given here: https://github.com/myhealthconnectsociety/project-healthcare/pull/109

**Comment by aryan0931** [comment]
hello @codecakes can you assign this issue to me ?

**Comment by devXpraAddy** [comment]
@codecakes I would like to resolve this issue under C4GT community, please assign this issue to me

**Comment by Hafsa-Ahmadi** [comment]
Hello @codecakes I would like to contribute. Could you please assign this issue to me?

**Comment by Ritesh-Thorve** [comment]
hi @codecakes close this issue if it is solved.

**Comment by Atulya12345** [comment]
Hi **@codecakes** ,

I'd love to take up issue #57 as part of my C4GT contributions. This task aligns well with my prior experience working on secure API integrations and environment-based configuration. **I’ve completed the implementation, verified everything via Swagger UI, and raised a PR for the same — please refer to:**

https://github.com/myhealthconnectsociety/project-healthcare/pull/179

As a small extension, I can also add Pytest-based e2e tests and enhance the README by documenting how FROM_ORIGIN_HEADER_SECRET should be configured in local and CI environments, **along with how it integrates with the FromHeader binder and middleware validation flow.**

Looking forward to your review — please consider assigning this issue to me officially. Happy to refine if needed!

Thanks!

**Comment by pikender** [comment]
@codecakes I have raised a [PR](https://github.com/myhealthconnectsociety/project-healthcare/pull/182).

Application changes are working as expected but not able to add the tests as test setup is not working.

Please have a look and suggest next steps on same

**Comment by utkarshgupta2804** [comment]
Hey @codecakes,
Is this issue still open?
